pkg/pipeline/lifecycle.go

```diff
@@ -142,21 +152,35 @@ func (s *Service) stopWithReason(ctx context.Context, pl *Instance, reason error
 	return err
 }
 
+func (s *Service) stopForceful(ctx context.Context, pl *Instance) error {
+	s.logger.Debug(ctx).Str(log.PipelineIDField, pl.ID).Msg("force stopping pipeline")
+	pl.t.Kill(cerrors.New("force stop"))
+	for _, n := range pl.n {
+		if node, ok := n.(stream.ForceStoppableNode); ok {
+			// stop all pub nodes
+			s.logger.Trace(ctx).Str(log.NodeIDField, n.ID()).Msg("force stopping node")
+			node.ForceStop(ctx)
+		}
+	}
+	return nil
+}
+
 // StopAll will ask all the pipelines to stop gracefully
 // (i.e. that existing messages get processed but not new messages get produced).
 func (s *Service) StopAll(ctx context.Context, reason error) {
 	for _, pl := range s.instances {
 		if pl.Status != StatusRunning {
 			continue
 		}
-		err := s.stopWithReason(ctx, pl, reason)
+		err := s.stopGraceful(ctx, pl, reason)
 		if err != nil {
 			s.logger.Warn(ctx).
 				Err(err).
 				Str(log.PipelineIDField, pl.ID).
 				Msg("could not stop pipeline")
 		}
 	}
+	// TODO stop pipelines forcefully after timeout if they are still running
```

Comment:
is this out of scope? or is it not that high of a priority right now?

Reply:
I didn't want to increase the scope of this issue since the PR was already quite big. We can tackle this in a separate PR, we will need to rethink how we want this to work, e.g. do we want to force stop pipelines after a timeout? Should that timeout be configurable? Or maybe it should be triggered by the second SIGINT signal (currently Conduit is just killed by the second SIGINT).